File: SDKs/amazonaws.com/clouddirectory/2016-05-10/java/lib/src/main/java/openapisdk/models/shared/NotNodeException.java
package openapisdk.models.shared;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonInclude.Include;

/**
 * NotNodeException
 * Occurs when any invalid operations are performed on an object that is not a node, such as calling <code>ListObjectChildren</code> for a leaf node object.
**/
public class NotNodeException {
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("Message")
    public String message;
    public NotNodeException withMessage(String message) {
        this.message = message;
        return this;
    }
}

File: SDKs/amazonaws.com/clouddirectory/2016-05-10/python/src/sdk/models/shared/notnodeexception.py
import dataclasses
from typing import Optional
from dataclasses_json import dataclass_json
from sdk import utils


@dataclass_json
@dataclasses.dataclass
class NotNodeException:
    r"""NotNodeException
    Occurs when any invalid operations are performed on an object that is not a node, such as calling <code>ListObjectChildren</code> for a leaf node object.
    """
    
    message: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('Message') }})
    
